clients/cobol-lsp-vscode-extension/src/services/Settings.ts

```diff
@@ -229,10 +229,10 @@ export class SettingsService {
     documentUri: string,
     dialectType: string,
   ): Promise<string[]> {
-    const pgPaths = await loadProcessorGroupCopybookPaths(
-      documentUri,
-      dialectType,
-    );
+    const pgPaths = (
+      await loadProcessorGroupCopybookPaths(documentUri, dialectType)
+    ).filter((element) => typeof element == "string");
```

Comment:
The function returns `string[]`, why the filter?

Reply:
it can be an object array. Filtering is to obtain only locals.

Reply:
```
export async function loadProcessorGroupCopybookPaths(
  documentUri: string,
  dialectType: string,
): Promise<string[]>
```

If you are not getting string[], there is something wrong.

Reply:
loadProcessorGroupCopybookPaths returns object and not modified in this scope of this pr, however for sure it can be refactored.

Reply:
This still needs to be addressed.

Reply:
Would something like this  work  ? https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/pull/2758/commits/8998716429ce7a5960f6064dbd29f1fa14da26ee

Reply:
No, that's the same problem moved a call down the stack - this time it is the `loadProcessorGroupSettings` which returns `string[]`. Fix the underlying issue.

Reply:
`loadProcessorGroupSettings` returns  `Promise<T>` not `string[]` . I'm struggling to follow the issue.

Reply:
Yes, and `T` in that case is `string[]`

Reply:
no, it is a `T` that might be string[], or array of objects or array of hybrid data types ...

I'm all open to suggestions for better way

Reply:
My suggestion is to actually investigate the root cause of the problem.

Based on a quick research the problem is likely one of the following type assertions 
https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/blob/3383129d0905ac83fe4a2c78a2a10c57d4feeb72/clients/cobol-lsp-vscode-extension/src/services/ProcessorGroups.ts#L273-L278

Reply:
I don't see a problem related with this PR but related with this one : https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/pull/2451
Refactoring the consuming of processor groups settings can be handled in separate PR as It requires changes to be made irrevalant to this PR's scope

Reply:
Does https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/pull/2758/commits/355f1e709baa9edadfe496b52b97ae7b863a48fa this makes sense ?